src/Data/Text/IO/Utf8.hs

```diff
@@ -0,0 +1,44 @@
+-- |
+-- Module      : Data.Text.IO.Utf8
+-- Copyright   : (c) 2009, 2010 Bryan O'Sullivan,
+--               (c) 2009 Simon Marlow
+-- License     : BSD-style
+-- Maintainer  : bos@serpentine.com
+-- Portability : GHC
+--
+-- Efficient UTF-8 support for text I\/O.
+module Data.Text.IO.Utf8
+    (
+      readFile
+    , writeFile
+    , appendFile
+    ) where
+
+import Prelude hiding (readFile, writeFile, appendFile)
+import Control.Exception (evaluate)
+import Control.Monad ((<=<))
+import Data.ByteString (ByteString)
+import qualified Data.ByteString as B
+import Data.Text (Text)
+import Data.Text.Encoding (decodeUtf8, encodeUtf8)
+
+decodeUtf8IO :: ByteString -> IO Text
+decodeUtf8IO = evaluate . decodeUtf8
+
+encodeUtf8IO :: Text -> IO ByteString
+encodeUtf8IO = evaluate . encodeUtf8
+
+-- | The 'readFile' function reads a file and returns the contents of
+-- the file as a string.  The entire file is read strictly, as with
+-- 'getContents'.
+readFile :: FilePath -> IO Text
+readFile = decodeUtf8IO <=< B.readFile
```

Comment:
This operation is partial in the event that the given file *isn't* encoded in UTF-8. I'd personally prefer the signature of `readFile` to return an `Either` with a decode error if that happens. However, I acknowledge that that's contrary to the rest of the API of this package - the convention would be to define `readFile' :: FilePath -> IO (Either DecodeError Text)`.

Reply:
As you said, this doesn't match the convention of the rest of the package. Also, I generally don't like putting `Either` inside of `IO`, because IO can throw other errors too.